### DISCUSSION TITLE
Amethyst Geodes should probably only generate in the Overworld

### DISCUSSION THREAD

**Discussion by GregoriusT** [Roadhog360/Et-Futurum-Requiem#290] [Improvements]
So I have been testing some random Twilight Forest Stuff in GT6, while also happening to have a somewhat freshly compiled EFR in my Setup, and I noticed floating Geodes in Twilight Forest.

The way the Geodes generate indiscriminately causes problems with the Structures that generate in there should they happen to clash with each other, not to mention Twilight Forest has a Terrain height of 32, so I ended up seeing quite a lot of floating Geodes too.

I would recommend either:

- making sure they only generate in the Overworld, leaving ALL other Dimensions untouched unless whitelisted, or
- making sure they dont generate in Twilight Forest, or
- at least to make sure their height is not above 32, or
- check if there is Stone where Ores can generate inside of around a sample set of points.

**Comment by Roadhog360** [comment]
This is what the blacklisting functionality is for.

**Comment by GregoriusT** [comment]
Well yes ofcourse it is, but this is not something the end user does, this is something the dev has to decide because sensible defaults are important. ;)

And as i said there is more than that one option you mentioned, dont just skip over all of them. ;)

**Reply by Roadhog360** [reply]
I suppose you're right. I'll try to implement a better way to check *where* the geode can appear, ever since I ported the new geodes from 1.17 a few generation issues have admittedly popped up.
I'm not sure how I can guarantee TF to be blacklisted since iirc the user can change the dimension ID, but since many dimensions contain stone and default worldgen elements in them I think it's a reasonable approach to attempt to give the geode a better placement check.

I think this should be moved to an issue

**Comment by DarianLStephens** [comment]
Also, is it possible to switch the blacklist to a whitelist?
I remember getting quite annoyed at some mods when they only have a blacklist for worldgen. (Tinkers' Construct slime islands...)

**Comment by Roadhog360** [comment]
Yeah
